OpenRA.Mods.Common/Traits/Repairable.cs

```diff
@@ -91,6 +91,8 @@ public void ResolveOrder(Actor self, Order order)
 				var target = Target.FromOrder(self.World, order);
 				self.SetTargetLine(target, Color.Green);
 
+				TryCallTransport(self, target, movement);
+
 				self.CancelActivity();
```

Comment:
Took me a while, but I remember why now: if I have a single `nextActivity` object and use it for both sets of activities, like I just did (yes, it looks much better), the moment the carryall is close enough to the unit to lock it for pickup, it cancels the current set of activities. Because this is C#, that is the same set that is stored in `Carryable.afterLandActivity`. So that also gets cancelled, resulting in the unit remaining idle when dropped off at the repair pad.